components/Article/EditLink/EditLink.stories.tsx

```diff
@@ -0,0 +1,41 @@
+import {
+  exampleRelativePath,
+  i18nMockDataEnglish,
+  i18nMockDataNonEnglish,
+} from './mockDataConstants';
+import { LocaleProvider } from './../../../providers/localeProvider';
+import EditLink from './index';
+import type { Meta as MetaObj, StoryObj } from '@storybook/react';
+
+type Story = StoryObj<typeof EditLink>;
+type Meta = MetaObj<typeof EditLink>;
+
+export const Edit: Story = {
+  args: {
+    relativePath: exampleRelativePath,
+  },
+  decorators: [
+    Story => (
+      <LocaleProvider i18nData={i18nMockDataEnglish}>
```

Comment:
We already have a constants file for all this. Check `.storybook/constants.js`.

Also we already inject i18n data by default on Storybook. No need to do this again.

Reply:
`EditLink` logic relies heavily on the selected language.
Initially I wanted to make it explicit that we have language logic here by mocking the `i18nData` which is shared between Story and tests.

Did you mean something like this?
```jsx
<LocaleProvider i18nData={{ pageProps.i18nData..., currentLocale: { code: 'en' } }}>
<LocaleProvider i18nData={{ pageProps.i18nData..., currentLocale: { code: 'xx' } }}>
```
If yes, then I agree it is cleaner. Then for tests, I'll mock it inside `index.test.tsx`.

@MaledongGit Since it's related, let's discuss here.
I think let's keep it to English for now for both implementation simplicity and universality. Actually if I don't know the translated meaning, I'll assume they're also the same as "Edit on GitHub" which is incorrect.

Reply:
Honestly, we should not test different languages on the Storybook.

Otherwise, we need to add a global toggle on Storybook that allows you to switch languages.

And I guess the best way of doing so https://storybook.js.org/docs/react/essentials/controls would somewhat be to extend the preview or something.

But for now, I instead remove this from here. Also, having nested Providers isn't a good approach here.

Reply:
> But for now, I instead remove this from here.

Sorry, I am just making sure that I got it correct.
So, you are suggesting to remove it, right?

I am okay with it since tests will cover the testing and if other people want to preview it on Storybook, they can just change the `currentLocale.code` value in `.storybook/constants.js`.

Reply:
Yup, I'm fine with removing :)